src/configs.py

```diff
@@ -43,10 +47,24 @@ def load(cls, requirer: Any) -> "DatabaseConfig":
         )
 
 
+@dataclass
+class StartTLSConfig:
+    enabled: bool = True
+    tls_key: Path = SERVER_KEY
+    tls_cert: Path = SERVER_CERT
+
+    @classmethod
+    def load(cls, charm: CharmBase) -> "StartTLSConfig":
```

Comment:
I would prefer it if we passed only the config object here (maybe as a dict). No need for the whole charm object.

Reply:
Yep, good point.